code/modules/pool/pool.dm

```diff
@@ -0,0 +1,298 @@
+		var/mob/living/carbon/human/F = user
+		var/datum/species/SS = F.dna.species
+		if(MOB_ROBOTIC in SS.inherent_biotypes)  //ZAP goes the IPC!
+			zap = 2 //You can protect yourself from water damage with thick clothing.
+		if(F.head && istype(F.head, /obj/item/clothing))
+			var/obj/item/clothing/CH = F.head
+			if (CH.clothing_flags & THICKMATERIAL) //Skinsuit should suffice! But IPCs are robots and probably not water-sealed.
+				zap --
+		if(F.wear_suit && istype(F.wear_suit, /obj/item/clothing))
+			var/obj/item/clothing/CS = F.wear_suit
+			if (CS.clothing_flags & THICKMATERIAL)
+				zap --
+	if(zap > 0)
+		user.emp_act(zap)
+		user.emote("scream") //Chad coders use M.say("*scream")
+		do_sparks(zap, TRUE, user)
+		to_chat(user, "<span class='userdanger'>WARNING: WATER DAMAGE DETECTED!</span>")
+		SEND_SIGNAL(user, COMSIG_ADD_MOOD_EVENT, "robotpool", /datum/mood_event/robotpool)
+	else
+		if(!check_clothes(user))
+			SEND_SIGNAL(user, COMSIG_ADD_MOOD_EVENT, "pool", /datum/mood_event/poolparty)
+			return
+		SEND_SIGNAL(user, COMSIG_ADD_MOOD_EVENT, "pool", /datum/mood_event/poolwet)
+
+//Largely a copypaste from shower.dm. Checks if the mob was stupid enough to enter a pool fully clothed. We allow masks as to not discriminate against clown and mime players.
+/turf/open/indestructible/sound/pool/proc/check_clothes(mob/living/carbon/human/H)
+	if(!istype(H)) //Don't care about non humans.
+		return FALSE
+	if(iscatperson(H))
+		return FALSE
+	if(H.wear_suit && (H.wear_suit.clothing_flags & SHOWEROKAY))
+		// Do not check underclothing if the over-suit is suitable.
+		// This stops people feeling dumb if they're showering
+		// with a radiation suit on.
+		return FALSE
+
+	. = FALSE
+	if(H.wear_suit && !(H.wear_suit.clothing_flags & SHOWEROKAY))
+		. = TRUE
+	else if(H.w_uniform && !(H.w_uniform.clothing_flags & SHOWEROKAY))
+		. = TRUE
+	else if(H.head && !(H.head.clothing_flags & SHOWEROKAY))
+		. = TRUE
+
+/obj/effect/turf_decal/pool
+	name = "Pool siding"
+	icon = 'icons/obj/pool.dmi'
+	icon_state = "poolborder"
+
+/obj/effect/turf_decal/pool/corner
+	icon_state = "bordercorner"
+
+/obj/effect/turf_decal/pool/innercorner
+	icon_state = "innercorner"
+
+//Pool machinery
+
+/obj/structure/pool_ladder
+	name = "Pool ladder"
+	desc = "Click this to get out of a pool quickly."
+	icon = 'icons/obj/pool.dmi'
+	icon_state = "ladder"
+	pixel_y = 12
+
+GLOBAL_LIST_EMPTY(pool_filters)
+
+/obj/machinery/pool_filter
+	name = "Pool filter"
+	desc = "A device which can help you regulate conditions in a pool. Use a <b>wrench</b> to change its operating temperature, or hit it with a reagent container to load in new liquid to add to the pool."
+	icon = 'icons/obj/pool.dmi'
+	icon_state = "poolfilter"
+	pixel_y = 12 //So it sits above the water
+	idle_power_usage = IDLE_POWER_USE
+	var/id = null //change this if youre an annoying mapper who wants multiple pools per area.
+	var/list/pool = list()
+	var/desired_temperature = 300 //Room temperature
```

Comment:
```suggestion
	var/desired_temperature = 293 //Room temperature
```
Assuming this is in Kelvin, this is wrong.

Reply:
Room temperature in kelvin ranges from 283 to 303 according to https://www.nist.gov/pml/weights-and-measures/si-units-temperature Should be fine, I think this is only used for checking anyway